cli/src/commands/examples.rs

```diff
@@ -0,0 +1,234 @@
+    let latest_release = examples_repo
+        .releases()
+        .get_latest()
+        .await
+        .context("Can't access the examples releases. Please retry later")?;
+
+    let example_asset = if let Some(example) = &example_opts.name {
+        // Check if the example exists
+        let example_lowercase = example.to_lowercase();
+        latest_release
+            .assets
+            .into_iter()
+            .find(|a| a.name.trim_end_matches(".zip") == example_lowercase)
+            .ok_or(anyhow!(
+                "Unknown example {}. Use `restate examples` to navigate the list of examples.",
+                example_lowercase
+            ))?
+    } else {
+        // Ask the example to download among the available examples
+        let mut languages = parse_available_examples(latest_release.assets);
+        let language_selection = Select::new()
+            .with_prompt("Which language?")
+            .items(&languages)
+            .interact()
+            .unwrap();
+        let example_selection = Select::new()
+            .with_prompt("Which example?")
+            .items(&languages[language_selection].examples)
+            .interact()
+            .unwrap();
+
+        languages
+            .remove(language_selection)
+            .examples
+            .remove(example_selection)
+            .asset
+    };
+
+    download_example(examples_repo, example_asset).await
+}
+
+struct Language {
+    display_name: String,
+    examples: Vec<Example>,
+}
+
+impl fmt::Display for Language {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.display_name)
+    }
+}
+
+struct Example {
+    display_name: String,
+    asset: Asset,
+}
+
+impl fmt::Display for Example {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.display_name)
+    }
+}
+
+fn parse_available_examples(assets: Vec<Asset>) -> Vec<Language> {
+    let mut languages_map = HashMap::new();
+
+    for asset in assets {
+        // Zip names have the format [language]-[example_name].zip
+
+        let asset_name = asset.name.clone();
+        let mut split_asset_name = asset_name.splitn(2, '-');
+
+        // First part is language
+        let language = if let Some(lang) = split_asset_name.next() {
+            lang.to_string()
+        } else {
+            // Bad name, just ignore it
+            continue;
+        };
+        let language_display_name = capitalize(&language);
+
+        // Second is example name
+        let example_display_name = if let Some(example) = split_asset_name.next() {
+            example
+                .to_string()
+                .from_case(Case::Kebab)
+                .to_case(Case::Title)
+        } else {
+            // Bad name, just ignore it
+            continue;
+        };
+        let example_display_name = example_display_name.trim_end_matches(".zip").to_string();
+
+        languages_map
+            .entry(language)
```

Comment:
```suggestion
            .entry(language.to_owned())
```
You'll need this if you decided to use the let-else syntax above.